lib/screens/home/widgets/appbar.dart

```diff
@@ -0,0 +1,246 @@
+import 'package:flutter/material.dart';
+
+import 'task_list.dart';
+
+class HomeAppBar extends StatefulWidget implements PreferredSizeWidget {
+  const HomeAppBar({
+    super.key,
+    required this.requireNotDone,
+    required this.switchRequireNotDone,
+    required this.requireFav,
+    required this.switchRequireFav,
+    required this.taskListKey,
+  }) : preferredSize = const Size.fromHeight(kToolbarHeight);
+
+  final bool requireNotDone;
+  final Function switchRequireNotDone;
```

Comment:
```suggestion
  final VoidCallback switchRequireNotDone;
```
Указывайте полную сигнатуру методов